.github/actions/download-pr-data-artifact/action.yaml

```diff
@@ -0,0 +1,43 @@
+name: "Download PR number saved as an artifact"
+description: |
+  This action can be used together with save-pr-as-artifact custom action which uploads the PR number as an artifact.
+  This action downloads the artifact to retrieve the PR number.
+outputs:
+  "pr_number":
+    value: ${{ steps.set-pr-number.outputs.pr_number }}
+    description: The PR number downloaded from the artifact
+runs:
+  using: "composite"
+  steps:
+    - name: Download artifact
+      uses: actions/github-script@v6
+      with:
+        script: |
+          let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.payload.workflow_run.id,
+          });
+          let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+            return artifact.name == "pr_number"
+          })[0];
```

Comment:
What if there is no matching artifacts? Can the `[0]` return a failure? I believe an `array out of bounds` error may be likely here.

Reply:
This is supposed to be used in conjunction with the save-pr-as-artifact. it's ok for the workflow to fail if that's not the case.